php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -460,7 +461,7 @@ public ParameterTree PARAMETER() {
         b.zeroOrMore(ATTRIBUTE_GROUP()),
         b.optional(VISIBILITY_MODIFIER()),
         b.optional(b.token(PHPKeyword.READONLY)),
-        b.optional(DECLARED_TYPE()),
+        b.optional(b.firstOf(INTERSECTION_TYPE(),DECLARED_TYPE())),
```

Comment:
Since an intersection type is a declared type, I think it's better if we move `INTERSECTION_TYPE()` into `DECLARED_TYPE()` which would then become `b.firstOf(UNION_TYPE(), INTERSECTION_TYPE(), TYPE())`